### ISSUE TITLE
7.5.3 API Gateway REST error on push - CREATE_FAILED APIGatewayAuthStack TemplateURL must be a supported URL.

### ISSUE THREAD

**Issue by fpodlaha** [aws-amplify/amplify-cli#9124]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.15.5

### Amplify CLI Version

7.5.3

### What operating system are you using?

Windows

### Amplify Categories

api

### Amplify Commands

add, push

### Describe the bug

Since updating from 6.3.0 to 7.5.3 (and upgrading all sections) I have a problem with the REST API Gateway.  When Updating or Adding a REST api, and choosing to "Restrict API Access? Yes" with "Authenticated users only" (or "Authenticated users or guests"); I get an error when pushing the changes ("amplify push").  

> **CREATE_FAILED APIGatewayAuthStack AWS::CloudFormation::Stack Sun Nov 28 2021 17:13:41 GMT-0500 (Eastern Standard Time) TemplateURL must be a supported URL.**

No matter what I try with the api: creating a new one, updating, creating a new path, create new function, update function, etc.; whenever I choose to "Restrict API Access" - the "amplify push" throws that error (and of course, rollsback).

If I choose not to restrict access "Restrict API Access? No" - then "amplify push" completes successfully.  Of course, then my functions are wide open.

Other options chosen for my function: language=python, it uses storage and environmental variables.  Changing the language (to NodeJs) or removing storage/env variables has no affect.

### Expected behavior

I expect "amplify push" to succeed just the same as if I chose not to Restrict API access (as it did before in prior versions).

### Reproduction steps

**amplify api add**
? Select from one of the below mentioned services: **REST**
√ Would you like to add a new path to an existing REST API: (y/N) · **no**
√ Provide a friendly name for your resource to be used as a label for this category in the project: · **testapi**
√ Provide a path (e.g., /book/{isbn}): · **/somepath**
√ Choose a Lambda source · Create a new Lambda function
? Provide an AWS Lambda function name: **somelambdafunction**
? Choose the runtime that you want to use: **NodeJS**
? Choose the function template that you want to use: **Hello World**

Available advanced settings:
- Resource access permissions
- Scheduled recurring invocation
- Lambda layers configuration
- Environment variables configuration
- Secret values configuration

? Do you want to configure advanced settings? No
? Do you want to edit the local lambda function now? No
Successfully added resource somelambdafunction locally.

Next steps:
Check out sample function code generated in <project-dir>/amplify/backend/function/somelambdafunction/src
"amplify function build" builds all of your functions currently in the project
"amplify mock function <functionName>" runs your function locally
To access AWS resources outside of this Amplify app, edit the C:\Users\Frank\Desktop\BrightLens\Code\planogram\amplify\backend\function\somelambdafunction\custom-policies.json
"amplify push" builds all of your local backend resources and provisions them in the cloud
"amplify publish" builds all of your local backend and front-end resources (if you added hosting category) and provisions them in the cloud
✅ Succesfully added the Lambda function locally
√ Restrict API access? (Y/n) · **yes**
√ Who should have access? · **Authenticated users only**
√ What permissions do you want to grant to Authenticated users? · **create, read, update, delete**
√ Do you want to add another path? (y/N) · **no**
✅ Successfully added resource testapi locally

✅ Some next steps:
"amplify push" will build all your local backend resources and provision it in the cloud
"amplify publish" will build all your local backend and frontend resources (if you have hosting category added) and provision it in the cloud

**amplify push**

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by Goose39** [comment]
I can confirm that I have been struggling with the same issue and have also followed the same steps you have described.

**Comment by jhockett** [comment]
Hi @fpodlaha and @Goose39, this appears to be a Windows bug that's been fixed in 7.5.4. Can you upgrade the CLI and see if that solves the issue?

**Comment by Goose39** [comment]
@jhockett Thank you! I can confirm that this solved the issue for me.
